### PR TITLE
Add parameter validation for ECDSA module

### PR DESCRIPTION
**Pull request by hanno-arm**
__Summary:__ Based on #1643, this PR adds parameter validation for the ECDSA module.

### CODE REVIEW

File: include/mbedtls/ecdsa.h [hunk 1]
```diff
@@ -425,36 +483,51 @@ int mbedtls_ecdsa_genkey( mbedtls_ecdsa_context *ctx, mbedtls_ecp_group_id gid,
  *
  * \see             ecp.h
  *
- * \param ctx       The ECDSA context to set.
- * \param key       The EC key to use.
+ * \param ctx       The ECDSA context to set. This must be initialized.
+ * \param key       The EC key to use. This must be initialized and hold
+ *                  a private-public key pair or a public key. In the former
+ *                  case, the ECDSA context may be used for signature creation
```
[the remark this answers is not in the document]
Comment (hanno-arm) [reply include/mbedtls/ecdsa.h]: Fixed.

**Comment by mpg** [comment]
CI only failed in known-intermittent GnuTLS fragmentation test, so as good as a PASS.

**Comment by sbutcher-arm** [comment]
Reviewed, approved, passes CI, but needs rebasing.

Removing the 'needs: preceding PR' label, as PR #1643 has already been merged.

**Comment by mpg** [comment]
@sbutcher-arm I don't think this branch needs rebasing, it has no conflicts and was based on a recent enough version to pass CI.

**Comment by sbutcher-arm** [comment]
@mpg - Yes! Thanks for changing the base. That has removed the #1643 PR commits from this one.

The Jenkins CI has passed with only a failure of `DTLS proxy: 3d, gnutls server, fragmentation, nbio`, which is known to be flakey. The PR has been reviewed, approved, and passes CI, so labelling as 'ready to merge'.

**Comment by sbutcher-arm** [comment]
Jenkins CI failing on BSD timing test and `DTLS proxy: 3d, gnutls server, fragmentation` which is known to be flakey, so accepting as a pass.

The 'ready to merge' label wasn't removed, so leaving it in place.

**Pull request merged**